### DISCUSSION TITLE
runOnDemand - Pipe command outputs

### DISCUSSION THREAD

**Discussion by davidtaddei** [bluenviron/mediamtx#2944] [Questions]
### Question

Server version: 

I have a scenario where I need to pre-process a stream of bytes before they are passed to ffmpeg. Ordinarily I'd do the following to take the output of a pre-processor and feed it to ffmpeg:

`python main.py | ffmpeg -i pipe:0`

In this case the main.py pre-processor is writing to stdout and when run in a normal shell this works perfectly. Running a similar command directly or wrapped in a bash script as part of the mediamtx `runOnDemand` config items results in mediamtx capturing the stdout from `main.py` and logging it rather than sending onward to ffmpeg.

Does anyone know of a way to use stdin/stdout in a `runOnDemand` config entry?

**Comment by aler9** [comment]
Hello, you can use bash commands and pipes in `runOnDemand` by instantiating bash:

```yml
paths:
  mypath:
    runOnDemand: bash -c 'python main.py | ffmpeg -i pipe:0'
```

[a comment by an automated account is not shown]
